File: atlas-backend/app/schemas/course.py
"""
Course schemas
"""

from pydantic import BaseModel
from typing import Optional


class CourseCreate(BaseModel):
    """Schema for creating a new course"""
    name: str
    code: Optional[str] = None
    term: Optional[str] = None
    color: str = "#1F77B4"  # Default blue color
    instructor: Optional[str] = None
    description: Optional[str] = None


class CourseUpdate(BaseModel):
    """Schema for updating a course"""
    name: Optional[str] = None
    code: Optional[str] = None
    term: Optional[str] = None
    color: Optional[str] = None
    instructor: Optional[str] = None
    description: Optional[str] = None


class CourseResponse(BaseModel):
    """Schema for course response"""
    id: int
    name: str
    code: Optional[str] = None
    term: Optional[str] = None
    color: str
    instructor: Optional[str] = None
    description: Optional[str] = None
    created_at: str
    updated_at: Optional[str] = None
    
    class Config:
        from_attributes = True

File: atlas_study_app/lib/core/models/course.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'course.freezed.dart';
part 'course.g.dart';

@freezed
class Course with _$Course {
  const factory Course({
    required int id,
    required String name,
    String? code,
    String? term,
    @Default('#1F77B4') String color,
    String? instructor,
    String? description,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _Course;

  factory Course.fromJson(Map<String, dynamic> json) => _$CourseFromJson(json);
}

@freezed
class CourseCreate with _$CourseCreate {
  const factory CourseCreate({
    required String name,
    String? code,
    String? term,
    @Default('#1F77B4') String color,
    String? instructor,
    String? description,
  }) = _CourseCreate;

  factory CourseCreate.fromJson(Map<String, dynamic> json) => _$CourseCreateFromJson(json);
}

@freezed
class CourseUpdate with _$CourseUpdate {
  const factory CourseUpdate({
    String? name,
    String? code,
    String? term,
    String? color,
    String? instructor,
    String? description,
  }) = _CourseUpdate;

  factory CourseUpdate.fromJson(Map<String, dynamic> json) => _$CourseUpdateFromJson(json);
}
